### DISCUSSION TITLE
[Feature] Add function to backup and restore hidden difficulties

### DISCUSSION THREAD

**Discussion by lamda05** [ppy/osu#28658] [Ideas]
For some reason I don’t know, some loved beatmapsets have blank difficulties, usually named “marathon”. These difficulties are unplayable, and as they have no hit objects, osu!lazer would throw a error when they get selected. So, I got them all hidden.

However, after I mistakenly hid a normal [_no pun intended_] difficulty, I had to clear all the hidden difficulties, including these blank difficulties that had to be manually hidden again.

It would be good if there be a function to backup and restore hidden difficulties, for example, exporting a list of hidden difficulties, which can be imported after.

**Comment by bdach** [comment]
> For some reason I don’t know, some loved beatmapsets have blank difficulties, usually named “marathon”.

As far as I'm aware most cases of this is people skirting beatmap file limit sizes by including a throwaway difficulty which increases allowable size for the entire set.

**Comment by Doksha** [comment]
What lazer really needs on the beatmap download page is a "force update/download specific difficulties" button without having to delete and re-download the entire set. About once a week I will either accidentally overwrite a main difficulty, or hide a main difficulty, and then have to delete the entire mapset just to download it again.
also sometimes the map just bugs and it would be nice to just be able to force download the map.
